ablySpec/RealtimeClientChannel.swift

```diff
@@ -186,6 +187,56 @@ class RealtimeClientChannel: QuickSpec {
             // RTL6
             describe("publish") {
 
+                // RTL6a
+                it("should encode messages in the same way as the RestChannel") {
+                    let data = ["value":1]
+                    var restEncodedMessage: ARTMessage?
+                    var realtimeEncodedMessage: ARTMessage?
+
+                    let rest = ARTRest(options: AblyTests.commonAppSetup())
+                    let restChannel = rest.channels.get("test")
+
+                    let restBlock: @convention(block) (AspectInfo, ARTMessage) -> Void = { info, message in
+                        let invocation = info.originalInvocation()
+                        invocation.invoke()
```

Comment:
I'm not sure how this aspects thing works, but if `message` here is the original, unencoded message that is passed to `encodeMessageIfNeeded`, take into account that `ARTMessage.encode` returns a _copy_ of the ARTMessage instance, and thus this `message` won't be mutated by `invocation.invoke()`, and you're assigning the unencoded version to `restEncodedMessage`.

Ignore this if this `message` is what's returned by `encodeMessageIfNeeded`. (But then I don't understand what `invocation.invoke()` is invoking.)